### PR TITLE
Upload fixes

### PR DESCRIPTION
**Pull request by azhao12345**
we should validate the captcha before processing the upload to protect better from potential attacks

### CODE REVIEW

File: router/upload.go [hunk 1]
```diff
@@ -109,7 +113,6 @@ func (f *UploadForm) ExtractInfo(r *http.Request) error {
 		return ErrInvalidTorrentCategory
 	}
 
-
```
Comment (sfan5) [inline router/upload.go]: the 2 blank lines here are supposed to split the function into logical units...

Comment (azhao12345) [inline router/upload.go]: I just ran go fmt and it stripped the extra whitespace.  Not sure if trying to keep non-standard whitespace is sustainable at all.

Comment (majestrate) [inline router/upload.go]: if `go fmt` removed it it should stay removed

Comment (sfan5) [inline router/upload.go]: go sucks ass confirmed

**Pull request merged**